aws/resource_aws_cognito_identity_pool_roles_attachment.go

```diff
@@ -141,17 +143,19 @@ func resourceAwsCognitoIdentityPoolRolesAttachmentRead(d *schema.ResourceData, m
 	log.Printf("[DEBUG] Reading Cognito Identity Pool Roles Association: %s", d.Id())
 
 	ip, err := conn.GetIdentityPoolRoles(&cognitoidentity.GetIdentityPoolRolesInput{
-		IdentityPoolId: aws.String(d.Get("identity_pool_id").(string)),
+		IdentityPoolId: aws.String(d.Id()),
 	})
 	if err != nil {
-		if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == "ResourceNotFoundException" {
+		if isAWSErr(err, cognitoidentity.ErrCodeResourceNotFoundException, "") {
 			log.Printf("[WARN] Cognito Identity Pool Roles Association %s not found, removing from state", d.Id())
 			d.SetId("")
 			return nil
 		}
 		return err
 	}
 
+	d.Set("identity_pool_id", d.Id())
```

Comment:
since the API call to get the identity pool roles returns the id info, we should set this attribute as `d.Set("identity_pool_id", ip.IdentityPoolId)`

Reply:
Thanks! Missed that